test/code/plugins/squid_log_parser_plugintest.rb

```diff
@@ -0,0 +1,61 @@
+## squid_log_parser_plugintest
+## Created by Alessandro Cardoso
+## 
+module Fluent
+require 'fluent/parser'
+require 'test/unit'
+require_relative '../../../source/code/plugins/squidlogparser.rb'
+
+class SquidLogParserTestRuntimeError < Fluent::SquidLogParser
+  def log_error(text)
+    raise text
+  end
+end
+
+class SquidLogParserTest < Test::Unit::TestCase
+  class << self
+    def startup
+      @@squidlog = Fluent::SquidLogParser.new(SquidLogParserTestRuntimeError.new)
+    end
+
+    def shutdown
+      #no op
+    end
+  end
+
+  def test_parse
+
+    text = '1486686388.768      0 10.1.1.4 TCP_MISS/200 2965 GET cache_object://localhost/5min - HIER_NONE/- text/plain'  
+    record = @@squidlog.parse(text)
+    assert_equal record['HostName'], 'contoso.SquidServer1'
+    assert_equal record['ResourceID'], 'Squid'
+   
+    text = '1486686229.592     97 10.1.1.4 TCP_MISS/301 483 open http://portal.azure.com/ - HIER_DIRECT/13.77.0.21 text/html'
+    record = @@squidlog.parse(text)
+    assert_equal record['HostName'], 'contoso.SquidServer1'
+    assert_equal record['ResourceID'], 'Squid'
+
+    text = '1486686488.885      0 10.1.1.4 TCP_DENIED/403 8175 POST http://www.testurl.com/ - HIER_NONE/- text/html'
+    record = @@squidlog.parse(text)
+    assert_equal record['HostName'], 'contoso.SquidServer1'
+    assert_equal record['ResourceID'], 'Squid'
+
+    text = '1486686167.769   5010 10.1.1.4 TCP_MISS/304 386 GET http://ctldl.windowsupdate.com/msdownload/update/v3/static/trustedr/en/disallowedcertstl.cab? - HIER_DIRECT/72.247.223.179 application/octet-stream'
+    record = @@squidlog.parse(text)
+    assert_equal record['HostName'], 'contoso.SquidServer1'
+    assert_equal record['ResourceID'], 'Squid' 
+  
+    text = '1486686448.842     23 10.1.1.4 TCP_MISS/400 1513 open http://cloudtidings.com/ - HIER_DIRECT/192.0.78.25 text/html' 
+    record = @@squidlog.parse(text)
+    assert_equal record['HostName'], 'contoso.SquidServer1'
+    assert_equal record['ResourceID'], 'Squid'
+   
+  
+    text = '1486686629.120     23 10.1.1.4 TCP_MISS/400 1513 open http://cloudtidings.com/ - HIER_DIRECT/192.0.78.25 text/html'
+    record = @@squidlog.parse(text)
+    assert_equal record['HostName'], 'contoso.SquidServer1'
+    assert_equal record['ResourceID'], 'Squid'
+
+  end
+
+end
```

Comment:
Please comment your "end" keywords with what they are ending, as you did in squidstats.rb